### PR TITLE
Fix application modal and routing

### LINKED ISSUE #784: change url once filter is applied on `/applications`
### Reproducibility

 - [ ] This issue is reproducible
 - [ ] This issue is not reproducible

### Steps to Reproduce
1. 
2. 

### Severity/Priority
 - [ ] Critical
 - [ ] High
 - [ ] Medium
 - [ ] Low

### Checklist

- [ ] I have read and followed the project's code of conduct.
- [ ] I have searched for similar issues before creating this one.
- [ ] I have provided all the necessary information to understand and reproduce the issue.
- [ ] I am willing to contribute to the resolution of this issue.

### PR DESCRIPTION
**Pull request by vinit717**
## Issue Ticket Number
closes #784, #788, #792 

## Description

- Automatic Modal Display: When a superuser navigates to /application?id=<applicationId>, the modal will automatically be displayed.

- Button State Management: If the application has already been accepted or rejected, the corresponding action button will be disabled.

- Dynamic URL Updates: The URL will update dynamically based on the filters applied, reflecting the selected filter parameters.

- API Limit Adjustment: The API has been updated to fetch 6 applications at a time, ensuring that all available space in the application cards is utilized effectively.

- Enable applications button on dashboard page

### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots

https://github.com/user-attachments/assets/75dde7f1-27bd-437d-987e-b721bb2d3e58

Screenshot from 2024-08-27 23-33-25

### CODE REVIEW

**Comment by vinit717** [comment]
For test fix and enhancement here is the issue ticket
https://github.com/Real-Dev-Squad/website-dashboard/issues/791

File: __tests__/home/home.test.js [hunk 1]
```diff
@@ -523,7 +523,7 @@ describe('Home Page', () => {
       (el) => el.getAttribute('href'),
       applicationButton,
     );
-    expect(applicationButtonHref).toBe('/applications');
+    expect(applicationButtonHref).toBe('/applications/index.html');
```
Comment (yesyash) [inline __tests__/home/home.test.js:526]: - why is this required?

File: applications/script.js [hunk 2]
```diff
@@ -73,6 +81,10 @@ function closeApplicationDetails() {
   applicationDetailsModal.classList.add('hidden');
   backDropBlur.style.display = 'none';
   document.body.style.overflow = 'auto';
+
+  if (applicationId) {
+    window.location.href = '/applications';
+  }
```
Comment (yesyash) [inline applications/script.js:87]: - why is this check required?

**Review by Achintya-Chatterjee — APPROVED** [review]

File: index.html [hunk 3]
```diff
@@ -138,8 +138,8 @@
       >
       <a
         id="application-button"
-        href="/applications"
-        class="action-button element-display-remove"
+        href="/applications/index.html"
```
Comment (yesyash) [inline index.html:141]: - why is this required does navigating to application not render the index.html by default?

File: __tests__/home/home.test.js (hunk 1 above)
Comment (vinit717) [reply __tests__/home/home.test.js:526]: No removed

File: applications/script.js (hunk 2 above)
Comment (vinit717) [reply applications/script.js:87]: Because when super user go to this route applications/?id=<applicationId> then a particular application modal open and this check is in close modal function by this if the particular modal closes it redirect me to all application page

File: applications/style.css [hunk 4]
```diff
@@ -157,7 +158,7 @@ body {
 .application-card {
   border-radius: 15px;
   box-shadow: var(--elevation-1);
-  padding: 15px;
+  padding: 24px;
```
Comment (joyguptaa) [inline applications/style.css:161]: NIT : would be better if we use rem instead of px

**Review by joyguptaa — APPROVED** [review]

File: applications/script.js [hunk 5]
```diff
@@ -73,9 +81,16 @@ function closeApplicationDetails() {
...
 
-function openApplicationDetails(application) {
+function openApplicationDetails(application, renderById) {
+  if (!renderById) {
+    const currentUrlParams = new URLSearchParams(window.location.search);
+    currentUrlParams.append('id', application.id);
+    const applicationByIdUrl = '/applications/?' + currentUrlParams.toString();
+    window.history.replaceState(window.history.state, '', applicationByIdUrl);
+  }
   currentApplicationId = application.id;
```
Comment (Achintya-Chatterjee) [inline applications/script.js:94]: `openApplicationDetails(application, renderById)` function is overloaded with responsibilities—managing both the application data rendering and URL state. 



Split the function into smaller functions, one for managing the URL state and another for rendering the application details.

File: applications/script.js [hunk 6]
```diff
@@ -170,15 +185,35 @@ function openApplicationDetails(application) {
...
+    applicationAcceptButton.disabled = true;
+    applicationAcceptButton.style.cursor = 'not-allowed';
+    applicationAcceptButton.classList.add('disable-button');
+  } else {
+    applicationRejectButton.disabled = false;
+    applicationRejectButton.style.cursor = 'pointer';
+    applicationRejectButton.classList.remove('disable-button');
+
+    applicationAcceptButton.disabled = false;
+    applicationAcceptButton.style.cursor = 'pointer';
```
Comment (Achintya-Chatterjee) [inline applications/script.js:209]: The logic that enables/disables buttons based on application status (accept/reject) is buried within the `openApplicationDetails` function. 


Extract this conditional logic into a helper function, improving readability and making it easier to manage button states.

**Review by Achintya-Chatterjee — CHANGES_REQUESTED** [review]
@samarpan1738 please

File: applications/script.js (hunk 5 above)
Comment (samarpan1738) [reply applications/script.js:94]: Good point. Extracted 2 methods to add and remove query param from the url and update it

File: applications/style.css (hunk 4 above)
Comment (samarpan1738) [reply applications/style.css:161]: Initially I has used rem here. 
But then I thought of these questions
1. Do we need `rem` for padding? 
2. Do we need to adapt to user preferences for the padding?

Acc. to my opinion, we don't need rem here

**Review by Achintya-Chatterjee — APPROVED** [review]
